packages/client/src/runtime/core/request/PrismaPromise.ts

```diff
@@ -8,31 +26,34 @@ export interface PrismaPromise<A> extends Promise<A> {
    * Extension of the original `.then` function
    * @param onfulfilled same as regular promises
    * @param onrejected same as regular promises
-   * @param txId GUID for interactive txs
+   * @param transaction interactive transaction options
    */
   then<R1 = A, R2 = never>(
     onfulfilled?: (value: A) => R1 | PromiseLike<R1>,
     onrejected?: (error: unknown) => R2 | PromiseLike<R2>,
-    txId?: string,
+    transaction?: InteractiveTransactionOptions,
   ): Promise<R1 | R2>
 
   /**
    * Extension of the original `.catch` function
    * @param onrejected same as regular promises
-   * @param txId GUID for interactive txs
+   * @param transaction interactive transaction options
    */
-  catch<R = never>(onrejected?: ((reason: any) => R | PromiseLike<R>) | undefined | null, txId?: string): Promise<A | R>
+  catch<R = never>(
+    onrejected?: ((reason: any) => R | PromiseLike<R>) | undefined | null,
+    transaction?: InteractiveTransactionOptions,
+  ): Promise<A | R>
 
   /**
    * Extension of the original `.finally` function
    * @param onfinally same as regular promises
-   * @param txId GUID for interactive txs
+   * @param transaction interactive transaction options
    */
-  finally(onfinally?: (() => void) | undefined | null, txId?: string): Promise<A>
+  finally(onfinally?: (() => void) | undefined | null, transaction?: InteractiveTransactionOptions): Promise<A>
 
   /**
    * Called when executing a batch of regular tx
-   * @param txId for regular tx ids
+   * @param transaction transaction options for regular tx
    */
-  requestTransaction?(txId: number, lock?: PromiseLike<void>): PromiseLike<unknown>
+  requestBatchTransaction?(transaction: BatchTransactionOptions, lock?: PromiseLike<void>): PromiseLike<unknown>
```

Comment:
Took the liberty to rename this method to be more specific about which of the 2 kinds of transactions it requests. It is not a part of public API, but let me know if you still consider this rename dangerous, I'll revert.

Reply:
I've been wanting to even remove this or hide it but some people rely on it so we decided not to do so, although it's not documented as public API :hear_no_evil:

Reply:
Ok, reverting the change then